src/c_api/query_result.cpp

```diff
@@ -75,39 +89,63 @@ bool kuzu_query_result_has_next_query_result(kuzu_query_result* query_result) {
     return static_cast<QueryResult*>(query_result->_query_result)->hasNextQueryResult();
 }
 
-kuzu_query_result* kuzu_query_result_get_next_query_result(kuzu_query_result* query_result) {
+kuzu_state kuzu_query_result_get_next_query_result(kuzu_query_result* query_result,
```

Comment:
It might be helpful to check `hasNextQueryResult` and return an error if no next.